test/std/containers/sequences/array/array.comparison/greater_g.pass.cpp

```diff
@@ -0,0 +1,63 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "oneapi_std_test_config.h"
+
+#include _ONEAPI_STD_TEST_HEADER(array)
+
+#include <iostream>
+
+#include "support/utils.h"
+
+namespace test_ns = _ONEAPI_TEST_NAMESPACE;
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr auto sycl_write = sycl::access::mode::write;
+
+bool
+kernel_test()
+{
+    sycl::queue deviceQueue = TestUtils::get_test_queue();
+    bool ret = false;
+    sycl::range<1> numOfItem{1};
+    {
+        sycl::buffer<bool, 1> buffer1(&ret, numOfItem);
+        deviceQueue.submit([&](sycl::handler& cgh) {
+            auto ret_access = buffer1.get_access<sycl_write>(cgh);
+            cgh.single_task<class KernelTest>([=]() {
+                const size_t len = 5;
+                typedef test_ns::array<int, len> array_type;
+                array_type a = {{0, 1, 2, 3, 4}};
+                array_type b = {{0, 1, 2, 3, 4}};
+                array_type c = {{0, 1, 2, 3, 7}};
+                ret_access[0] = (!(a > b));
+                ret_access[0] &= (c > a);
+            });
+        });
+    }
+    return ret;
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+#if TEST_DPCPP_BACKEND_PRESENT
+    auto ret = kernel_test();
+    EXPECT_TRUE(ret, "");
```

Comment:
IMHO, an error message should be meaningful, in order to quick understand what and where is wrong.
That comment regards to other similar places in this PR.